core/src/SYCL/Kokkos_SYCL_Parallel_Range.hpp

```diff
@@ -94,17 +94,14 @@ class Kokkos::Impl::ParallelFor<FunctorType, ExecPolicy,
     const Kokkos::Experimental::SYCL& space = m_policy.space();
     Kokkos::Experimental::Impl::SYCLInternal& instance =
         *space.impl_internal_space_instance();
-    Kokkos::Experimental::Impl::SYCLInternal::IndirectKernelMemory& kernelMem =
-        *instance.m_indirectKernel;
+    using IndirectKernelMem =
+        Kokkos::Experimental::Impl::SYCLInternal::IndirectKernelMem;
+    IndirectKernelMem& indirectKernelMem = instance.m_indirectKernelMem;
```

Comment:
Is there any good reason to spell out the type?
Rather than just
```C++
auto& indirectKernelMem = instance.m_indirectKernelMem;

Reply:
I'm just not a fan of `auto` in general, as it hides too much information from code readers.

Of course, the verbosity of long types also is annoying for code readers...